src/pybind/mgr/dashboard/controllers/docs.py

```diff
@@ -0,0 +1,284 @@
+                                   METH_ORDER.index(e.method.lower()))
+            for endpoint in endpoint_list:
+                if not endpoint.is_api and not all_endpoints:
+                    skip = True
+                    break
+
+                method = endpoint.method
+                params = []
+                params.extend([self._gen_param(p, 'path')
+                               for p in endpoint.path_params])
+                params.extend([self._gen_param(p, 'query')
+                               for p in endpoint.query_params])
+
+                if method.lower() in ['post', 'put']:
+                    body_params = self._gen_body_param(endpoint.body_params)
+                    if body_params:
+                        params.append(body_params)
+
+                # uncomment the lines below as soon as we have support for
+                # basic auth
+                # if endpoint.is_secure:
+                #     security = '{ "basicAuth": [] }'
+                # else:
+                #     security = ''
+                security = ''
+
+                methods += """
+                "{}": {{
+                    "tags": ["{}"],
+                    "summary": "",
+                    "consumes": [
+                        "application/json"
+                    ],
+                    "produces": [
+                        "application/json"
+                    ],
+                    "parameters": [{}],
+                    {},
+                    "security": [{}]
+                }},
+                """.format(method.lower(), endpoint.group, ",".join(params),
+                           self._gen_responses_descriptions(method), security)
+            if not skip:
+                paths += """
+                "{}": {{ {} }},
+                """.format(path[len(baseUrl):], methods[:-1])
+
+        if not baseUrl:
+            baseUrl = "/"
+        spec = """
+        {{
+            "swagger": "2.0",
+            "info": {{
+                "description": "Please note that this API is not an official \
+Ceph REST API to be used by third-party applications. It's primary purpose is \
+to serve the requirements of the Ceph Dashboard and is subject to change at \
+any time. Use at your own risk.",
+                "version": "v1",
+                "title": "Ceph-Dashboard REST API"
+            }},
+            "host": "{}",
+            "basePath": "{}",
+            "tags": [{}],
+            "schemes": ["https"],
+            "paths": {{
+                {}
+            }},
+        }}
+        """.format(host, baseUrl, self._gen_tags(all_endpoints), paths[:-1])
+
+        return spec
+
+    @Endpoint(json_response=False)
+    def __call__(self, all_endpoints=False):
+        page = """
+        <!DOCTYPE html>
+        <html>
+        <head>
+            <meta charset="UTF-8">
+            <link href="https://fonts.googleapis.com/css?family=Open+Sans:400, \
```

Comment:
I don't think it is an good idea to reference an external URL like this. Consider the situation that a customer do not want to have external references because of security reasons. Our solution must run out of the box shipping all required dependencies.

Reply:
I'm not sure, we have to obey this here. This endpoint is not required for the core functionality. On the other hand, this might be a security issue, as we're leaking private info to the outside world here. We're exposing the referer via HTTP headers.

Reply:
Couldn't this be fixed by adding `rel="noreferrer"` to the link?